src/Common/src/TypeSystem/IL/Stubs/StructMarshallingThunk.Mangling.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.TypeSystem;
+
+namespace Internal.IL.Stubs
+{
+    public partial class StructMarshallingThunk : IPrefixMangledType
+    {
+        TypeDesc IPrefixMangledType.BaseType
+        {
+            get
+            {
+                return ManagedType;
+            }
+        }
+
+        string IPrefixMangledType.Prefix
+        {
+            get
+            {
+                if (ThunkType == StructMarshallingThunkType.ManagedToNative)
```

Comment:
A switch statement would be safer here. If we add more thunk types in the future, they'll fall into the Cleanup category silently.